staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -175,7 +175,7 @@ func Invalid(field *Path, value interface{}, detail string) *Error {
 func NotSupported(field *Path, value interface{}, validValues []string) *Error {
 	detail := ""
 	if validValues != nil && len(validValues) > 0 {
```

Comment:
any time quoting is involved, I'm biased to actually quote using `strconv.Quote`:

```
	quotedValues := make([]string, len(validValues))
	for i, v := range validValues {
		quotedValues[i] = strconv.Quote(v)
	}
	detail = "supported values: " + strings.Join(quotedValues, ", ")
```